packages/flutter/test/material/popup_menu_test.dart

```diff
@@ -2209,6 +2209,87 @@ void main() {
     await tester.pumpAndSettle();
     expect(find.text('foo'), findsOneWidget);
   });
+
+  testWidgets('The opened menu should follow if the button\'s position changed', (WidgetTester tester) async {
+    final GlobalKey buttonKey = GlobalKey();
+
+    Widget buildFrame(double width, double height) {
+      return MaterialApp(
+        home: Scaffold(
+          body:  SizedBox(
+            height: height,
+            width: width,
+            child: Center(
+              child: PopupMenuButton<int>(
+                child: SizedBox(
+                  key: buttonKey,
+                  height: 10.0,
+                  width: 10.0,
+                  child: const ColoredBox(
+                    color: Colors.pink,
+                  ),
+                ),
+                itemBuilder: (BuildContext context) => <PopupMenuEntry<int>>[
+                  const PopupMenuItem<int>(child: Text('-1-'), value: 1),
+                  const PopupMenuItem<int>(child: Text('-2-'), value: 2),
+                ],
+              ),
+            ),
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(buildFrame(100.0, 100.0));
+
+    // Open the menu.
+    await tester.tap(find.byKey(buttonKey));
+    await tester.pumpAndSettle();
+
+    //                 +--------+--------+ 100
+    //                 |        |        |
+    //                 |        | (50,50)|
+    //                 +--------+--------+
+    //                 |        |        |
+    //                 |        |        |
+    //             100 +--------+--------+
+    //
+    // The button is a rectangle of 10 * 10 size and is centered,
+    // so its top-left offset should be (45.0, 45.0).
+    Offset buttonOffset = tester.getTopLeft(find.byKey(buttonKey));
+    expect(buttonOffset, const Offset(45.0, 45.0));
+
+    // The top-left corner of the menu and button should be aligned.
+    Offset popupMenuOffset = tester.getTopLeft(find.byType(SingleChildScrollView));
+    expect(popupMenuOffset, buttonOffset);
+
+    // Keep the menu opened and re-layout the screen.
+    await tester.pumpWidget(buildFrame(200.0, 300.0));
+
+    //                 +-----------+-----------+ 200
```

Comment:
Wow, these look great! Did you handwrite these diagrams?

Reply:
haha, that is true.